cmd/apps/chart_app.go

```diff
@@ -131,12 +136,25 @@ before using the generic helm chart installer command.`,
 				setMap[k] = v
 			}
 		}
+		outputPath := path.Join(chartPath, chartName)
+		if helm3 {
+			err := helm3Upgrade(outputPath, chartRepoName, namespace,
+				"values.yaml",
+				defaultVersion,
+				setMap,
+				wait)
 
-		err = templateChart(chartPath, chartName, namespace, outputPath, "values.yaml", setMap)
-		if err != nil {
-			return err
-		}
+			if err != nil {
+				return err
+			}
 
+		} else {
+			outputPath = path.Join(chartPath, "chart/rendered")
+			err = templateChart(chartPath, chartName, namespace, outputPath, "values.yaml", setMap)
+			if err != nil {
+				return err
+			}
+		}
 		err = kubectl("apply", "--namespace", namespace, "-R", "-f", outputPath)
```

Comment:
This part should be only for helm2. The helm 3 upgrade don't need it and is causing the strange error.

Reply:
the `kubectl apply -f` is meant here.